lib/engine/step/special_track.rb

```diff
@@ -48,6 +53,27 @@ def ability(entity)
           entity == @round.just_sold_company
         ability || entity.abilities(:tile_lay, 'track')
       end
+
+      def check_connect(action)
+        company = action.entity
+        a = ability(company)
+        hex_ids = a.hexes
+        return unless a&.connect && hex_ids.size > 1
+
+        if company == @company
```

Comment:
doesn't this fail if the company doesn't lay the two tracks together? what if two companies are doing this separately

Reply:
Of the privates we have currently implemented that lay more than one track, all of them either block until all tiles are laid, or lose the ability if the track lays are interrupted, so I don't think we'll run into the issues you raised. I admit that's a weak defense though. Can you propose a different approach?

Reply:
i don't think your code does this properly